In [1]:
import re #importação do regex
# Leitura do arquivo csv
import pandas as pd
import numpy as np
pd.set_option('display.max_rows', None) #Não limita a exibição dos n°s de registros
pd.options.display.max_columns = None #Não limita a exibição das colunas dos DataFrames
pd.set_option('display.max_colwidth', None) #Não  limita a exibição do conteúdo das células
df_cat = pd.read_csv("TabelaCatalogo49.csv", low_memory=False)

In [2]:
print("Quantidades de linhas da tabela origem:",len(df_cat))

Quantidades de linhas da tabela origem: 4891


In [3]:
#Pego todos os registros que tem suporte igual a 'Fotografias'
df_cat_foto = df_cat[df_cat['suporte'] == 'Fotografias'] 
df_cat_foto

especificação  \
414                                                                                                                                                                                                               Revolução Mexicana: II - Representações artísticas -  Murais - Muralista: José Clemente Orozco. La Trinchera. (Local: Escuela Nacional Preparatória. Ciudad de México).   
415                                                                                                                                                                                                                                                                                    Revolução Mexicana: II - Representações artísticas -  Gravura - J. G. Posada - Calavera de Madero.   
416                                                                                                                                                                                                                                                                                    Revolução Mexicana: II - Representações artísticas -  Gravura - J. G. Posada - Calavera Zapatista.   
417                                                                                                                                                                                                                                    Revolução Mexicana: II - Representações artísticas -  Gravura - J. G. Posada - Combate de los federales contra los zapatistas. (Gravura em zinco).   
418                                                                                                                                                                                                                                                                   Revolução Mexicana: II - Representações artísticas -  Gravura - J. G. Posada - Emiliano Zapata. (Gravura em zinco).   
419                                                                                                                                                                                                                                                                           Revolução Mexicana: II - Representações artísticas -  Gravura - J. G. Posada - Entrada de Madero no México.   
420                                                                                                                                                                                                                                                                 Revolução Mexicana: II - Representações artísticas -  Murais - Muralista: Diego Rivera. La esclavitud em el trapiche.   
421                                                                                                                                                                                                                                                                     Revolução Mexicana: II - Representações artísticas -  Murais - Muralista: Diego Rivera. Féria del dia del mortos.   
422                                                                                                                                                                                                                                      Revolução Mexicana: II - Representações artísticas -  Murais - Muralista: Diego Rivera. Historia y perspectiva de México: las tres revoluciones.   
423                                                                                                                                                                                                                                                   Revolução Mexicana: II - Representações artísticas -  Murais - Muralista: Diego Rivera. Historia y perspectiva de México (detalhe).   
424                                                                                                                                                                          

In [4]:
print("Quantidade de registros em fotografias", len(df_cat_foto))

Quantidade de registros em fotografias 211


In [5]:
#Verificação dos conteúdos de cada coluna
for col in df_cat_foto.columns.tolist(): #pega as colunas do DataFrame e transforma em list
    col_unique = df_cat_foto[col].unique()
    if(not(len(col_unique) == 1 and col_unique[0] != col_unique[0])): #verifica se tem algum valor nas colunas, isto é se a coluna não tem apenas um valor, e se este valor não é NaN
        print("================>",col, len(col_unique)) #Destaca com ====> o nome da coluna e o tamanho de seus registros
        print( col_unique[0:4]) #exibe os 4 primeiros valores da coluna

================> especificação 205
['Revolução Mexicana: II - Representações artísticas -  Murais - Muralista: José Clemente Orozco. La Trinchera. (Local: Escuela Nacional Preparatória. Ciudad de México).'
 'Revolução Mexicana: II - Representações artísticas -  Gravura - J. G. Posada - Calavera de Madero.'
 'Revolução Mexicana: II - Representações artísticas -  Gravura - J. G. Posada - Calavera Zapatista.'
 'Revolução Mexicana: II - Representações artísticas -  Gravura - J. G. Posada - Combate de los federales contra los zapatistas. (Gravura em zinco).']
================> nreferencia 211
['Cx.2;E.2/NT 31' 'Cx.2;E.2/NT 32' 'Cx.2;E.2/NT 33' 'Cx.2;E.2/NT 34']
================> numero 209
[187. 188. 189. 190.]
================> local 22
[nan 'CAPH' 'Anfiteatro de História/USP' 'Bienal do Livro/SP']
================> titulo 30
['Revolução Mexicana: II - Representações artísticas - Murais, Gravuras.'
 'Revolução Mexicana: I - Representações fotográficas.'
 'Álbum de fotografias sobre Guarat

# Campos válidos de df_cat_foto 

    1. especificação 205 *
    2. nreferencia 211 *
    3. numero 209 *
    4. local 22 *
    5. titulo 30 *
    6. produção 5 *
    7. data 23 *
    8. quantidade 166
    9. Sistema 54 *
    *campos iguais ao geralBD.csv
    
## Campos possíveis de normalização/análise
    1. local
    2. data
    3. produção

In [6]:
#Atualizo o  DataFrame apenas com as colunas que possuem valores
df_cat_foto = pd.DataFrame(
    {'nreferencia':df_cat_foto['nreferencia'], 
     'especificação':df_cat_foto['especificação'], 
     'numero':df_cat_foto['numero'],
     'quantidade':df_cat_foto['quantidade'],
     'sistema':df_cat_foto['Sistema'],
     'titulo':df_cat_foto['titulo'],
     'local':df_cat_foto['local'],
     'data':df_cat_foto['data'],
     'produção':df_cat_foto['produção']
    })


In [7]:
df_cat_foto

nreferencia  \
414                                               Cx.2;E.2/NT 31   
415                                               Cx.2;E.2/NT 32   
416                                               Cx.2;E.2/NT 33   
417                                               Cx.2;E.2/NT 34   
418                                               Cx.2;E.2/NT 35   
419                                               Cx.2;E.2/NT 36   
420                                               Cx.2;E.2/NT 37   
421                                               Cx.2;E.2/NT 38   
422                                               Cx.2;E.2/NT 39   
423                                               Cx.2;E.2/NT 40   
424                                               Cx.2;E.2/NT 41   
425                                               Cx.2;E.2/NT 42   
426                                               Cx.2;E.2/NT 43   
427                                               Cx.2;E.2/NT 44   
428                                               Cx.2;E.2/NT 45   
429                                               Cx.2;E.2/NT 46   
430                                               Cx.2;E.2/NT 47   
1452                                               Cx.2;E.1/NT 1   
1453                                               Cx.2;E.1/NT 2   
2913                                                         115   
2923                                              Cx.1;E.1/NT 48   
2924                                              Cx.1;E.1/NT 49   
2925                                              Cx.1;E.1/NT 50   
2926                                              Cx.1;E.1/NT 51   
2927                                              Cx.1;E.1/NT 52   
2928                                              Cx.1;E.1/NT 53   
2929                                              Cx.1;E.1/NT 54   
2930                                              Cx.1;E.1/NT 55   
2933                                              Cx.1;E.1/NT 56   
2934                                              Cx.1;E.1/NT 57   
2935                                              Cx.1;E.1/NT 58   
2936                                              Cx.1;E.1/NT 59   
2937                                             Cx.1;E.2/NT 114   
2938                                             Cx.1;E.2/NT 113   
2939                                             Cx.1;E.2/NT 112   
2940                                             Cx.1;E.2/NT 111   
2941                                             Cx.1;E.2/NT 110   
2942                                             Cx.1;E.2/NT 109   
2943                                             Cx.1;E.2/NT 108   
2944                                             Cx.1;E.2/NT 107   
2945                                              Cx.1;E.2/NT 76   
2946                                              Cx.1;E.2/NT 75   
2947                                              Cx.1;E.2/NT 74   
2948                                              Cx.1;E.2/NT 73   
2949                                              Cx.1;E.2/NT 72   
2950                                              Cx.1;E.2/NT 71   
2951                                              Cx.1;E.2/NT 70   
2952                                              Cx.1;E.2/NT 69   
2953                                              Cx.1;E.2/NT 68   
2954                                              Cx.1;E.2/NT 67   
2955                                              Cx.1;E.2/NT 66   
2956                                              Cx.1;E.2/NT 65   
2957                                              Cx.1;E.2/NT 64   
2958                                              Cx.1;E.2/NT 63   
2959                                              Cx.1;E.2/NT 62   
2960                                              Cx.1;E.2/NT 61   
2961                                              Cx.1;E.2/NT 60   
2962                                              Cx.1;E.3/NT 92   
2963                                     

In [8]:
#verifico os tipos de valores do DF
df_cat_foto.dtypes

nreferencia       object
especificação     object
numero           float64
quantidade        object
sistema           object
titulo            object
local             object
data              object
produção          object
dtype: object

In [11]:
df_cat_foto.groupby('quantidade')['quantidade'].agg([np.size])
#OK, quantidade não é inteiro mesmo

size
quantidade      
1              6
1/12           1
1/14           1
1/16           1
1/25           2
1/47           1
1/6            1
1/7            1
10             2
10/12          1
10/14          1
10/16          1
10/25          2
10/47          1
100            1
11/12          1
11/14          1
11/16          1
11/25          2
11/47          1
12             1
12/12          1
12/14          1
12/16          1
12/25          2
12/47          1
124            1
13             2
13/14          1
13/16          1
13/25          2
13/47          1
14/14          1
14/16          1
14/25          2
14/47          1
141            1
15/16          1
15/25          2
15/47          1
16/16          1
16/25          2
16/47          1
17/25          2
17/47          1
18/25          1
18/47          1
19/25          2
19/47          1
2/12           1
2/14           1
2/16           1
2/25           2
2/47           1
2/6            1
2/7            1
20             1
20/25          2
20/47          1
21             1
21/25          2
21/47          1
22             1
22/25          2
22/47          1
23             4
23/25          2
23/47          1
24             1
24/25          2
24/47          1
25             1
25/25          2
25/47          1
26             1
26/25          1
26/47          1
27             1
27/47          1
28             1
28/47          1
29             1
29/47          1
3              2
3/12           1
3/14           1
3/16           1
3/25           2
3/47           1
3/6            1
3/7            1
30             4
30/47          1
31             2
31/47          1
32/47          1
33/47          1
34             4
34/47          1
35/47          1
36             1
36/47          1
37             1
37/47          1
38             1
38/47          1
39/47          1
4/12           1
4/14           1
4/16           1
4/25           2
4/47           1
4/6            1
4/7            1
40/47          1
41/47          1
42/47          1
43/47          1
44             1
44/47          1
45             1
45/47          1
46/47          1
47/47          1
5/12           1
5/14           1
5/16           1
5/25           2
5/37           1
5/47           1
5/6            1
5/7            1
54             1
59             1
6              1
6/12           1
6/14           1
6/16           1
6/25           2
6/47           1
6/6            1
6/7            1
62             1
64             1
65             1
68             1
69             1
7/12           1
7/14           1
7/16           1
7/25           2
7/47           1
7/7            1
8/12           1
8/14           1
8/16           1
8/25           2
8/37           3
8/47           1
9              2
9/12           1
9/14           1
9/16           1
9/25           2
9/37           1
9/47           1

In [14]:
df_cat_foto.local = df_cat_foto.local.replace('CAPH', 'CAPH-USP')
df_cat_foto.local = df_cat_foto.local.replace('São Paulo/Anfiteatro de Geografia', 'São Paulo/Anfiteatro Geografia')
df_cat_foto.groupby('local')['local'].agg([np.size])

size
local                                                     
Anfiteatro de Convenções/USP                             2
Anfiteatro de História/USP                               2
Biblioteca do Depto.Geografia/USP                        1
Bienal do Livro/SP                                       2
CAPH-USP                                                33
Departamento de Ciências Sociais/USP                     1
Departamento de Geografia/USP                            2
Departamento de História/USP                             2
Departamento de Letras/USP                               1
IEB                                                      1
Prédio da Administração FFLCH/USP                        1
Prédio de Geografia e História                           2
Rio de Janeiro/RJ                                        1
Salão Nobre da FFLCH/USP                                 5
Salão Nobre da FFLCH/USP e Centro de Informática         1
São Paulo                                                1
São Paulo/Anfiteatro Geografia                           2
São Paulo/Salão Nobre- FFLCH                            20
São Paulo/Salão Nobre-FFLCH/Centro de Convenções-USP     1

In [18]:
df_cat_foto.data = df_cat_foto.data.replace('s/d', None)
df_cat_foto.groupby('data')['data'].agg([np.size])

size
data               
01-30/04/1986    25
05/07/1922        1
07/06/1911        1
1937/1979         1
1945              1
1973              1
1985              2
1986              4
1987              1
1988              2
1989              7
1990              7
1991              1
1992              5
1994              2
1996              1
1997              5
1998              6
1999              3
22/03/2001        1
26/04/2001        1

In [19]:
df_cat_foto.groupby('produção')['produção'].agg([np.size])

size
produção                    
CAPH                      43
Col/9 x 12,5 cm            1
Eduardo José Afonso       25
Pavan Nagae Associados     1

In [21]:
#vou trabalhar com a col nreferencia
s_referencia = df_cat_foto.nreferencia
s_referencia[s_referencia.duplicated()]#pego as referencias repetidas
#Não há, então esta tabela já está pronta pro merge

Series([], Name: nreferencia, dtype: object)

In [22]:
df_cat_foto.to_csv("fotografia_tabela49.csv", index=False, encoding='UTF-8' )